src/webhook/validation/activegate.go

```diff
@@ -69,3 +73,28 @@ func missingActiveGateMemoryLimit(dv *dynakubeValidator, dynakube *dynatracev1be
 func memoryLimitSet(resources corev1.ResourceRequirements) bool {
 	return resources.Limits != nil && resources.Limits.Memory() != nil
 }
+
+func exclusiveSyntheticCapability(dv *dynakubeValidator, dk *dynatracev1beta1.DynaKube) string {
+	if dk.IsSyntheticActiveGateEnabled() &&
+		len(dk.Spec.ActiveGate.Capabilities) > 1 {
+		log.Info(
+			"requested dynakube has the synthetic active gate capability accompanied with others",
+			"name", dk.Name,
+			"namespace", dk.Namespace)
+		return fmt.Sprintf(errorJoinedSyntheticActiveGateCapability, syntheticlessCapabilities(dk))
+	}
+	return ""
+}
+
+func syntheticlessCapabilities(dk *dynatracev1beta1.DynaKube) string {
+	collected := []byte{'['}
+	for _, c := range dk.Spec.ActiveGate.Capabilities {
+		if c != dynatracev1beta1.SyntheticCapability.DisplayName {
+			if len(collected) > 1 {
+				collected = append(collected, ' ')
+			}
+			collected = append(collected, string(c)...)
+		}
+	}
+	return string(append(collected, ']'))
```

Comment:
```suggestion
	const separator = ' '

	collected := strings.Builder{}
	collected.WriteRune('[')

	for idx, c := range dk.Spec.ActiveGate.Capabilities {
		if c != dynatracev1beta1.SyntheticCapability.DisplayName {
			collected.WriteString(c)
			if idx < len(dk.Spec.ActiveGate.Capabilities)-1 {
				collected.WriteRune(separator)
			}
		}
	}

	collected.WriteRune(']')
	return collected.String()
```

Reply:
Reconciliation happens relatively seldom therefore we prefer code to be maintenance-optimized instead of performance-optimized. 
But good catch with `fmt.doPrint`!